include/internal/refcount.h

```diff
@@ -134,13 +135,15 @@ static __inline int CRYPTO_DOWN_REF(volatile int *val, int *ret,
 static __inline int CRYPTO_UP_REF(volatile int *val, int *ret,
                                   ossl_unused void *lock)
 {
+    assert(sizeof(int) == sizeof(long));
```

Comment:
No, this should be a static assert, that is checked by the compiler at compile time.

This kind of assert is a runtime assert (and it's normally compiled out, and only present in "debug" builds)

Reply:
Hmm, I'm one of the people who removed it!

```
commit fc785a554cc37dfa94710b28ced45b03006f0300
Author: Matt Caswell <matt@openssl.org>
Date:   Wed Sep 13 10:31:46 2023 +0100

    Remove use of _Static_assert

    We had some use of the C11 _Static_assert feature which can cause some
    problems on some platforms. Everywhere we were using it, it is not really
    required so remove it.

    Fixes #22017

    Reviewed-by: Tomas Mraz <tomas@openssl.org>
    Reviewed-by: Richard Levitte <levitte@openssl.org>
    Reviewed-by: Tom Cosgrove <tom.cosgrove@arm.com>
    Reviewed-by: Paul Dale <pauli@openssl.org>
    (Merged from https://github.com/openssl/openssl/pull/22091)
```

Reply:
In which case, for now, this last commit can be removed - sorry

Reply:
Could something like:
```
#if LONG_MAX != INT_MAX
# error This cast is incorrect if long size is different from int size
#endif
```
be used instead?

Of course it will require to `#include "internal/numbers.h"` before and the PR will no longer be eligible for CLA: trivial IMO.

Reply:
> In which case, for now, the last commit ("Incorporated review comment - added assert") can be removed from this PR - sorry

Reverted